tests/integration/test_tzkt_reward_api.py

```diff
@@ -80,18 +81,24 @@ def test_get_rewards_for_cycle_map(self, address, cycle):
         )
         tzkt_rewards = tzkt_impl.get_rewards_for_cycle_map(cycle, RewardsType.ACTUAL)
 
+        # TODO: Investigate why rpc staking balance is not equal to tzstats or tzkt
```

Comment:
Could there be a reason why the staking balance are not matching between rpc and tzkt/tzstats?

Reply:
Yes, could you check if there are still discrepancies? There should not be any. Last time I checked, TZKT and RPC agreed, but tzstats had a discrepancy.

If there are none, remove this comment

Reply:
Currently the baking_rights are not queries correctly. I created an issue [here](https://gitlab.com/tezos/tezos/-/issues/3791). Is there  another way to get the baking_rights via rpc for a given historic cycle?

Reply:
I replied to the issue. Rights for a given cycle have to be queried at the right level. This level normally is calculated with `last_block_level_in_cycle`, so maybe there is a problem with this calculation. Can you give an example query where there is a discrepancy?